From 825ddb98252d63651ead70eed32dfe5537c72ac6 Mon Sep 17 00:00:00 2001
From: kai <249905534@qq.com>
Date: Thu, 27 Aug 2026 21:14:28 +0800
Subject: [PATCH 5/6] Revert "arm64: dts: qcom: oneplus3: A/B div1-clk as
 fixed-clock (DTS boot fix)"

This reverts commit 77b11ffd06fed3eaea31adcf38b57e105cb0b6f6.
---
 arch/arm64/boot/dts/qcom/msm8996-oneplus-common.dtsi | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8996-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/msm8996-oneplus-common.dtsi
index 0af2de70da..e852dc1391 100644
--- a/arch/arm64/boot/dts/qcom/msm8996-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8996-oneplus-common.dtsi
@@ -32,14 +32,12 @@ chosen {
 
 	clocks {
 		div1_mclk: div1-clk {
-			/* A/B: switch gpio-gate-clock -> fixed-clock to match the
-			 * bootable v100 DTB. The gpio-gate-clock node referencing
-			 * pm8994_gpios/rpmcc is the prime suspect for the early
-			 * fastboot return. */
-			compatible = "fixed-clock";
+			compatible = "gpio-gate-clock";
+			pinctrl-names = "default";
+			pinctrl-0 = <&audio_mclk>;
 			#clock-cells = <0>;
-			clock-frequency = <24576000>;
-			clock-output-names = "divclk1";
+			clocks = <&rpmcc RPM_SMD_DIV_CLK1>;
+			enable-gpios = <&pm8994_gpios 15 GPIO_ACTIVE_HIGH>;
 		};
 
 		divclk4: div4-clk {
-- 
2.53.0

